warp/src/raygun/mod.rs

```diff
@@ -394,7 +395,7 @@ pub struct Conversation {
     created: DateTime<Utc>,
     favorite: bool,
     modified: DateTime<Utc>,
-    settings: ConversationSettings,
+    permissions: HashMap<DID, Vec<GroupPermission>>,
```

Comment:
Do we want to have the permissions here in Conversation or do you feel we should split this into a separate function?

Reply:
I created a type alias for it `GroupPermissions` as changing the type from `HashMap` to `IndexMap` required a lot of changes. So if we have to change the type again, for example to make it into a struct with more features, we won't have to change it everywhere again.

I also moved `has_permissions` method to the `GroupPermissions` type alias using a trait.